src/packagedcode/golang.py

```diff
@@ -141,3 +147,27 @@ def build_gomod_package(gomod_data):
         homepage_url=homepage_url,
         dependencies=package_dependencies
     )
+
+
+def build_gosum_package(gosum_data):
+    """
+    Return a Package object from a go.mod file or None.
```

Comment:
> Return a Package object from a go.mod file or None.

That looks like copied from go.mod

Reply:
Also when do you return None?

Reply:
Please ignore this for now. I have just created a simple initial file to get review why it is not working :)